chsdi/templates/htmlpopup/cadastralwebmap.mako

```diff
@@ -78,4 +78,38 @@
     % else:
         <tr><td class="cell-left">${_('link to canton geoportal')}</td><td>${_('Canton has provided no link to portal')}</td></tr>
     % endif
+    <%
+        import requests
+        import time
+        param_coord = request.params.get('coord')
+        coord = param_coord.split(',')
+        lat = coord[0]
+        lon = coord[1]
+        geodata = "http://geodata01.admin.ch/order/jPqrueQazrt/av_pdf.igs?pos=" + lat + '/' + lon
+
+        def get_url_data(url, max_tries=10):
+            for n in range(max_tries):
+                    r3 = requests.get(url)
```

Comment:
There's no need to the whole pdf file here. Try to do a HEAD request (try with curl first, then find out how to do it with requests library)